### ISSUE TITLE
the model report on the paper is trained by 300 epoches or 500 epoches?

### ISSUE THREAD

**Issue by gaopengcuhk** [facebookresearch/detr#61]
In your README, it seems that the final model is trained by 300 epochs with a learning rate drop at 200 epochs. 

However, in the following link, it seems like 42.0 is trained by 500 epochs with a learning rate drop at 400 epochs.

Can you clarify?

python -m torch.distributed.launch --nproc_per_node=8 --use_env main.py \
    --lr_drop 400 --epochs 500 \
    --coco_path /path/to/coco

https://gist.github.com/szagoruyko/9c9ebb8455610958f7deaa27845d7918

**Comment by fmassa** [comment]
Hi,

The ablations in the paper were conducted on 300 epochs, while the models reported in table 1 (main results) were trained for 500 epochs.

I believe I have answered your question, and as such I'm closing the issue but let us know if you have further questions.

**Comment by gaopengcuhk** [comment]
The final performance is trained for 500 epoches by 16GPU with 4img/GPU for DETR/DETR-DC5/DETR-R101/DETR-DC5-R101?

**Comment by gaopengcuhk** [comment]
If I understand correctly, the performance for 150 epoch is 39.5, for 500 epoch is 42.0. What's the performance of 300 epochs? I guess all models should be trained with the same batch size.

**Comment by gaopengcuhk** [comment]
If I understand correctly, the performance for 300 epochs should be 40.6?

**Comment by fmassa** [comment]
Yes, the performance for 300 epochs is 40.6, from table 2 in the paper.

**Comment by szagoruyko** [comment]
@gaopengcuhk sorry for the confusion in the gist! the 150 epoch model we provide to simplify reproducibility on a single 8-gpu machine, it was trained with batch size 4, thus total 32, to fit into 16GB cards.
Both the ablation 300-epoch (40.6) and the final 500-epoch (42.0) models were trained on 4 nodes with 2 im/gpu, thus total 64.

I will update non-DC5 cmd lines to submitit, and move the gist to the repo.

**Comment by gaopengcuhk** [comment]
The batch/GPU for 150 epoch provided in the github is 4 img/GPU or 2img/GPU?

**Comment by twangnh** [comment]
@szagoruyko  same question, also, could you pls provide the pretrained model of 150epoch run?
